x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/WaterfallFlyout.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Location } from 'history';
+import React from 'react';
+import { SpanFlyout } from './SpanFlyout';
+import { TransactionFlyout } from './TransactionFlyout';
+import { IWaterfall } from './waterfall_helpers/waterfall_helpers';
+
+interface Props {
+  waterfallItemId?: string;
+  waterfall: IWaterfall;
+  location: Location;
+  toggleFlyout: ({ location }: { location: Location }) => void;
+}
+export const WaterfallFlyout: React.FC<Props> = ({
+  waterfallItemId,
+  waterfall,
+  location,
+  toggleFlyout
+}) => {
+  const currentItem = waterfall.items.find(item => item.id === waterfallItemId);
+
+  if (!currentItem) {
+    return null;
+  }
+
+  switch (currentItem.docType) {
+    case 'span':
+      const parentTransaction = waterfall.getTransactionById(
+        currentItem.parentId
+      );
```

Comment:
What do you think about just adding the parent to each waterfall item? I don't expect this to cost a lot in increased memory since it'll just be pointers to transactions that are already in memory.

Reply:
Forgot to say: this will allow us to remove `getTransactionById` which is good because it decreases the API of waterfall helpers.